libraries/ssh_crypto.rb

```diff
@@ -247,6 +247,8 @@ def valid_algorithms # rubocop:disable Metrics/CyclomaticComplexity, Metrics/Met
       when /10.10\./, /10.11\./, /10.12\./
         alg66
       end
+    when 'amazon'
+      alg = alg53
```

Comment:
Do we need to distinguish between versions? I assume newer versions support the same as RHEL 7

Reply:
@jonasduarte I checked this params on Amazon Linux AMI 2017.09.0.20170930 x86_64 HVM and this immages supports the alg66. I think we should distinguish between the version.

Reply:
@jonasduarte  can you add the different params also?